File: packages/PIPS/validation/PHRASE/matmul.f
      subroutine MATMUL(a, b, c)

      integer i, j, k
      integer mat1(1:1000, 1:1000), mat2(1:1000, 1:1000), 
     & mat3(1:1000, 1:1000)

      do 10 i = 1, 1000
         do 20 j = 1, 1000
c BEGIN_FPGA_MATMULHRE
            mat3(i, j) = 0
            do 30 k = 1, 1000
               mat3(i, j) = mat3(i, j) + mat1(i, k) * mat2(k, j)
 30   enddo
c END_FPGA_MATMULHRE
 20   enddo
 10   enddo

      return
      end

File: Paws/openmp/matmul.c
int matmul(int a[100][100], int b[100][100], int c[100][100])
{
  int i, j, k;

  for(i = 0; i < 101; i++)
    {
      for(j = 0; j < 101; j++)
	{
	  for(k = 0; k < 101; k++)
	    {
	      a[i][j] = a[i][j] + b[i][k] * c[k][j];
	    }
	}
    }

  return 0;
}

int main(int argc, char* args)
{
  int a[100][100], b[100][100], c[100][100];

  matmul(a, b, c);



  return 0;
}
